manifests/init.pp

```diff
@@ -77,6 +77,22 @@
   }
 
   if $timezone::params::package {
+    if $package_ensure == 'present' and $::osfamily == 'Debian' {
+      $_area = split($timezone, '/')
+      $area = $_area[0]
+      $_zone = split($timezone, '/')
+      $zone = $_zone[1]
+      exec { 'update_debconf area':
+        command => "echo tzdata tzdata/Areas select ${area} | debconf-set-selections",
+        unless  => "debconf-get-selections |grep -q -E \"^tzdata\\s+tzdata/Areas\\s+select\\s+${area}\"",
+        path    => $::path,
+      }
+      exec { 'update_debconf zone':
+        command => "echo tzdata tzdata/Zones/${area} select ${timezone} | debconf-set-selections",
+        unless  => "debconf-get-selections |grep -E \"^tzdata\\s+tzdata/Zones/${area}\\s+select\\s+${zone}\"",
+        path    => $::path,
+      }
+    }
```

Comment:
@cmurphy the problme is that when the tzdata package is updated, debconf may restore the original configuration. Puppet will fix it at next run, but it is not very clean.